controllers/openstackcluster_controller.go

```diff
@@ -129,7 +131,35 @@ func (r *OpenStackClusterReconciler) reconcileCluster(logger logr.Logger, cluste
 
 	logger.Info("Reconciling network components")
 	if openStackCluster.Spec.NodeCIDR == "" {
-		logger.V(4).Info("No need to reconcile network")
+		logger.V(4).Info("No need to reconcile network, searching network and subnet instead")
+
+		netOpts := networks.ListOpts(openStackCluster.Spec.Network)
+		networkList, err := networkingService.GetNetworksByFilter(&netOpts)
+		if err != nil && len(networkList) == 0 {
+			return reconcile.Result{}, errors.Errorf("failed to find network: %v", err)
```

Comment:
do we allow the filter to be none? if so will this return error breaks the whole reconcile but actually we can continue?

Reply:
If neither the CIDR nor the Network Filter is configured it would break here. But I think this makes sense. It should be possible to fullfill all cases with those two properties. I think all other cases are easier to reason about if we can enforce here that we have a network and a subnet in the status. If not it just gets even more complicated.

Reply:
ok, let's get this in and if we have further question, we can work on that